File: app/src/main/java/com/lucasurbas/counter/ui/explore/di/ExploreFragmentComponent.java
package com.lucasurbas.counter.ui.explore.di;

import dagger.Subcomponent;
import com.lucasurbas.counter.app.di.FragmentModule;
import com.lucasurbas.counter.app.di.scope.FragmentScope;
import com.lucasurbas.counter.ui.explore.ExploreFragment;

@Subcomponent(modules = {
        FragmentModule.class, ExploreFragmentModule.class
})
@FragmentScope
public interface ExploreFragmentComponent {

    void inject(ExploreFragment fragment);
}

File: app/src/main/java/com/lucasurbas/counter/ui/explore/di/ExploreFragmentComponent.kt
package com.lucasurbas.counter.ui.explore.di

import com.lucasurbas.counter.app.di.FragmentModule
import com.lucasurbas.counter.app.di.scope.FragmentScope
import com.lucasurbas.counter.ui.explore.ExploreFragment
import dagger.Subcomponent

@Subcomponent(modules = arrayOf(FragmentModule::class, ExploreFragmentModule::class))
@FragmentScope
interface ExploreFragmentComponent {

    fun inject(fragment: ExploreFragment)
}
